apps/trading/client-pages/portfolio/account-history-container.tsx

```diff
@@ -0,0 +1,245 @@
+
+  const variables = useMemo(
+    () => ({
+      partyId: pubKey,
+      assetId: asset?.id || '',
+      accountTypes: accountType ? [accountType] : undefined,
+      dateRange:
+        range === 'All' ? undefined : { start: calculateStartDate(range) },
+    }),
+    [pubKey, asset, accountType, range]
+  );
+
+  const { data } = useAccountHistoryQuery({
+    variables,
+    skip: !asset || !pubKey,
+  });
+
+  return (
+    <div className="h-full w-full flex flex-col gap-8">
+      <div className="w-full flex flex-col-reverse lg:flex-row items-start lg:items-center justify-between gap-4 px-2">
+        <div className="flex items-center gap-4 shrink-0">
+          <DropdownMenu>
+            <DropdownMenuTrigger>
+              {accountType
+                ? `${
+                    AccountTypeMapping[
+                      accountType as keyof typeof Schema.AccountType
+                    ]
+                  } Account`
+                : t('Select account type')}
+            </DropdownMenuTrigger>
+            <DropdownMenuContent>
+              {Object.keys(Schema.AccountType).map((type) => (
+                <DropdownMenuItem
+                  key={type}
+                  onClick={() => setAccountType(type as Schema.AccountType)}
+                >
+                  {AccountTypeMapping[type as keyof typeof Schema.AccountType]}
+                </DropdownMenuItem>
+              ))}
+            </DropdownMenuContent>
+          </DropdownMenu>
+          <DropdownMenu>
+            <DropdownMenuTrigger>
+              {asset ? asset.symbol : t('Select asset')}
+            </DropdownMenuTrigger>
+            <DropdownMenuContent>
+              {assets.map((a) => (
+                <DropdownMenuItem key={a.id} onClick={() => setAsset(a)}>
+                  {a.symbol}
+                </DropdownMenuItem>
+              ))}
+            </DropdownMenuContent>
+          </DropdownMenu>
+        </div>
+        <div className="pt-1 justify-items-end">
+          <Toggle
+            id="account-history-date-range"
+            name="account-history-date-range"
+            toggles={dateRangeToggleItems}
+            checkedValue={range}
+            onChange={(e: ChangeEvent<HTMLInputElement>) =>
+              setRange(e.target.value)
+            }
+          />
+        </div>
+      </div>
+      <div className="h-5/6 px-4">
+        {asset && (
+          <AccountHistoryChart
+            data={data}
+            accountType={accountType}
+            asset={asset}
+          />
+        )}
+      </div>
+    </div>
+  );
+};
+
+export const AccountHistoryChart = ({
+  data,
+  accountType,
+  asset,
+}: {
+  data: AccountHistoryQuery | undefined;
+  accountType: Schema.AccountType;
+  asset: AssetFieldsFragment;
+}) => {
+  const theme = useContext(ThemeContext);
+  const values = useMemo(() => {
+    if (!data?.balanceChanges.edges.length) {
+      return [];
+    }
+
+    return compact(data.balanceChanges.edges)
```

Comment:
nit (non-blocking): not sure how big these results get, but there's 2 extra iterations happening which could be removed by having a reduce:

```
return data.balanceChanges.edges
  .reduce((acc, edge) => {
    if (edge.node.accountType === accountType) {
      acc.push({
        datetime: fromNanoSeconds(edge.node.timestamp),
        balance: Number(addDecimal(edge.node.balance, asset.decimals)),
      })
    }
    return acc
  })
  .reverse();
```

probably not a big impact thing, at least it's in a memo, so feel free to resolve this.

Reply:
updated in the last commit.